### ISSUE TITLE
Bug: newrelic_monitor_downtime throws an error when created with account_id specified in the provider

### ISSUE THREAD

**Issue by francis-shine** [newrelic/terraform-provider-newrelic#2546]
### Affected Resource(s)
- `newrelic_monitor_downtime`

### Terraform Configuration

```hcl
terraform {
  required_version = "~> 0.13.0"
  required_providers {
    newrelic = {
      source  = "newrelic/newrelic"
      version = "~> 3.28.1"
    }
  }
}

provider "newrelic" {
  account_id = local.env[var.account_type].newrelic_account_id
  region = "US"
}

resource "newrelic_monitor_downtime" "monitor_overnight_downtime" {
  count                 = local.env[var.account_type].overnight_shutdown ? 1 : 0
  name                  = "${var.environment} overnight monitor downtime"
  mode                  = "DAILY"

  monitor_guids         = [
    newrelic_synthetics_monitor.example_monitor.id
  ]

  start_time = "2024-01-01T22:00:00"
  end_time   = "2024-01-02T07:00:00"
  time_zone   = "Australia/Sydney"
}
```

### Actual Behavior
An error is thrown during `terraform apply` when no `account_id` is specified in the resource and the environment variable `NEW_RELIC_ACCOUNT_ID` is not specified:
```
Error: strconv.Atoi: parsing "": invalid syntax

  on infra/newrelic-monitor-downtime.tf line 1, in resource "newrelic_monitor_downtime" "monitor_overnight_downtime":
   1: resource "newrelic_monitor_downtime" "monitor_overnight_downtime" {
```

### Expected Behavior
The resource should be created using the `account_id` from the provider, as per the documentation

**Comment by pranav-new-relic** [comment]
Hi @francis-shine, thank you for reporting this. We've had the exact same issue reported two days ago (#2545 ) and I've added a clarification on our course of action in the [last comment in the issue](https://github.com/newrelic/terraform-provider-newrelic/issues/2545#issuecomment-1896004086). Can you please take a look at the issue I've linked here? Thanks!

**Comment by francis-shine** [comment]
Hi @pranav-new-relic 

Yes, this appears to be the same issue. It is resolved by manually specifying the `account_id` in the resource, but as mentioned in that comment it would be great if it was picked up from the `provider`! 

In lieu of this, a better error message would be greatly appreciated

Thanks!

**Comment by pranav-new-relic** [comment]
@francis-shine a fix to this has been released with v3.30.0 of the Terraform Provider released yesterday - this should allow `account_id` to default to that in the `provider{}` block if it isn't specified in the configuration of the resource. Thank you for reporting this.
